vendor/bat-native-ads/src/bat/ads/internal/ad_serving/ad_notifications/ad_notification_serving.cc

```diff
@@ -132,216 +150,56 @@ void AdServing::MaybeServeNextAd() {
-  BLOG(1, "Serve untargeted ad");
-
-  const std::vector<std::string> segments = {ad_targeting::kUntargeted};
-
-  database::table::CreativeAdNotifications database_table;
-  database_table.GetForSegments(
-      segments, [=](const Result result, const SegmentList& segments,
-                    const CreativeAdNotificationList& ads) {
-        EligibleAds eligible_ad_notifications(subdivision_targeting_,
-                                              anti_targeting_resource_);
-
-        const CreativeAdNotificationList eligible_ads =
-            eligible_ad_notifications.Get(ads, last_delivered_creative_ad_,
-                                          ad_events, history);
-
-        if (eligible_ads.empty()) {
-          BLOG(1, "No eligible ads found for untargeted segment");
-          BLOG(1, "Ad notification not served: No eligible ads found");
-          callback(Result::FAILED, AdNotificationInfo());
-          return;
-        }
-
-        MaybeServeAd(eligible_ads, callback);
-      });
-}
-
-void AdServing::MaybeServeAd(const CreativeAdNotificationList& ads,
-                             MaybeServeAdForSegmentsCallback callback) {
-  CreativeAdNotificationList eligible_ads = PaceAds(ads);
-  if (eligible_ads.empty()) {
-    BLOG(1, "Ad notification not served: No eligible ads found");
-    callback(Result::FAILED, AdNotificationInfo());
-    return;
-  }
-
-  BLOG(1, "Found " << eligible_ads.size() << " eligible ads");
-
-  const int rand = base::RandInt(0, eligible_ads.size() - 1);
-  const CreativeAdNotificationInfo ad = eligible_ads.at(rand);
-  MaybeDeliverAd(ad, callback);
-}
-
-CreativeAdNotificationList AdServing::PaceAds(
-    const CreativeAdNotificationList& ads) {
-  BLOG(2, ads.size() << " eligible ads before pacing");
-
-  AdPacing ad_pacing;
-  CreativeAdNotificationList paced_ads = ad_pacing.PaceAds(ads);
-
-  BLOG(2, paced_ads.size() << " eligible ads after pacing");
-
-  return paced_ads;
-}
-
-void AdServing::MaybeDeliverAd(const CreativeAdNotificationInfo& ad,
-                               MaybeServeAdForSegmentsCallback callback) {
-  AdNotificationInfo ad_notification;
-  ad_notification.type = AdType::kAdNotification;
-  ad_notification.uuid = base::GenerateGUID();
-  ad_notification.creative_instance_id = ad.creative_instance_id;
-  ad_notification.creative_set_id = ad.creative_set_id;
-  ad_notification.campaign_id = ad.campaign_id;
-  ad_notification.advertiser_id = ad.advertiser_id;
-  ad_notification.segment = ad.segment;
-  ad_notification.title = ad.title;
-  ad_notification.body = ad.body;
-  ad_notification.target_url = ad.target_url;
+      delay, base::BindOnce(&AdServing::MaybeServeAd, base::Unretained(this)));
+}
+
+bool AdServing::ServeAd(
+    const CreativeAdNotificationInfo& creative_ad_notification) const {
+  const AdNotificationInfo ad_notification =
+      BuildAdNotification(creative_ad_notification);
+
+  BLOG(1, "Serving ad notification:\n"
```

Comment:
same as above, though i guess we don't have an official policy for log level 1, only 0 (https://github.com/brave/brave-core/pull/8598/files)

Reply:
Raised https://github.com/brave/brave-browser/issues/16462